contracts/utils/deploy.js

```diff
@@ -244,6 +244,65 @@ const sendProposal = async (proposalArgs, description, opts = {}) => {
   log("Done");
 };
 
+/**
+ * Shortcut to create a deployment for hardhat to use
+ * @param {Object} options for deployment
+ * @param {Promise<Object>} fn to deploy contracts and return needed proposals
+ * @returns {Object} main object used by hardhat
+ */
+function deploymentWithProposal(opts, fn) {
+  const { deployName, dependencies } = opts;
+  const runDeployment = async (hre) => {
+    const tools = {
+      ethers,
+      deployWithConfirmation,
+    };
+    const proposal = await fn(tools);
+
+    const propDescription = proposal.name;
+    const propArgs = await proposeArgs(proposal.actions);
+
+    if (isMainnet) {
+      // On Mainnet, only propose. The enqueue and execution are handled manually via multi-sig.
+      log("Sending proposal to governor...");
+      await sendProposal(propArgs, propDescription);
+      log("Proposal sent.");
+    } else if (isFork) {
+      // On Fork we can send the proposal then impersonate the guardian to execute it.
+      log("Sending and executing proposal...");
+      await executeProposal(propArgs, propDescription);
+      log("Proposal executed.");
+    } else {
+      // Hardcoding gas estimate on Rinkeby since it fails for an undetermined reason...
+      const gasLimit = isRinkeby ? 1000000 : null;
+      for (const proposal of proposals) {
+        const { contract, signature, args } = proposal;
+        log(`Sending goverance action ${signature} to ${address}`);
```

Comment:
where is the address variable declared? I think you meant contract.address?

super nit: typo on governance